hilti/runtime/include/util.h

```diff
@@ -285,7 +286,48 @@ constexpr auto enumerate(T&& iterable) {
  * @param str string to expand
  * @return A UTF8 string with escape sequences expanded
  */
-std::string expandEscapes(std::string s);
+std::string expandUTF8Escapes(std::string s);
+
+namespace render_style {
+
+/**
+ * Flags specifying escaping style when rendering raw data for printing. The
+ * default style renders all non-printable characters as hex escapes (`\xNN`)
+ * and escapes backslashes with a second backslash. Any specified flags modify
+ * the default style accordingly.
+ */
+enum class Bytes {
+    Default = 0,                    /**< name for unmodified default style */
+    EscapeQuotes = (1U << 1U),      /**< escape double quotes with backslashes */
+    UseOctal = (1U << 2U),          /**< escape non-printables with `\NNN` instead of `\xNN` */
+    NoEscapeBackslash = (1U << 3U), /**< do not escape backslashes */
```

Comment:
It's a little confusing to me that some of these are "do X" but this one is "Don't do X" - I know there's some simplicity you get from this, but I'd prefer them all be one or the other.

My preference would be you pass in flags for what you want escaped, so this becomes `EscapeBackslash` and all logic flipped (and each time it is not passed in it gets passed in).